### ISSUE TITLE
Page number doesn't match

### ISSUE THREAD

**Issue by magiccube007** [mono424/studocu_unblocker#2]
As from the read me I think you are German I am just gonna write German, if thats not the case just tell me and i will translate.

Moin, hab auch die letzten paar wochen immer mal wieder des selbe wie du probiert und grad deine extension gesehen. Funktioniert soweil auch super, mir ist nur eine Kleinigkeit aufgefallen und zwar setzt du die pagenumber falsch. Aktuell kriegt man mit deiner version nur bis zur 9 seite auch die richtige seite angezeigt.

bg8 -> page 8,bg9 -> page9,bga -> page10,bgb -> page11,bgc->page12,bgd->page13,bge-> page14,bgf ->page15, bg10 -> page 16 usw. dann bg19, bg1a,bg1b usw.

Noch eine Sache zu dem Approach vielleicht hast du da eine Lösung für, der zieht nur alle images als eins, heißt wenn du eine file hat die nur ein bild pro seite ist funktioniert das ganze gut, sobald du aber eine file mit einem kleinen bild und nur text hast kriegst du nur das kleine bild.

Der Text wird über die requests geholt:

https://d3tvd1u91rr79.cloudfront.net/{id}/html/{id}1.page?Policy=&Signature=&Key-Pair-Id=

wäre z.b. das request format für die erste seite. Die hab ich noch nicht hinbekommen zu regenerieren, falls du da einen Tipp hast wäre ich sehr dankbar.

**Comment by mono424** [comment]
Hey erstmal danke für das Issue. Das mit den Seiten schaue ich mir mal an.
Das mit dem Content ist tricky habe mal alles, was ich herausgefunden hab in die Readme gepackt.
Bin über ideen dankbar! Vielleicht gibts irgendeine Möglichkeit die Signatur und Policy zu umgehen.

**Comment by magiccube007** [comment]
Ja, hab mir das readme gerade mal angeschaut dass is soweit eigentlich auch mein stand. Bin mir mittlerweile aber sehr sicher, dass die signatures für den text serverseitig generiert werden. Es gibt auch für jedes document eine "global" "QueryParam" (einfach mal in der response beim laden der seite schauen), ob diese zurück gegeben wird hängt vom document ab, also ob es ein premium document ist oder nicht. Wenn das document kein premium document ist kann man über den global parameter alles hollen auch direkt die webp ("url_front + "pages/page" + seitennummer + ".webp" + url_end") und spart sich somit, das umrechnen der Seiten zahl. 
In der App kann man sich diesen global Parameter für jedes Document dass in der App ist holen (bis jetzt eigentlich alle premium dokumente und die anderen waren auf der Seite), der endpoint hierfür wäre:
"https://www.studocu.com/app/v5/documents/" + str(id) +"/view?origin=Home%20Popular%20Documents&hl=de&hideAppBanner=1"
Problem ist bloß man braucht einen header "X-Firebase-AppCheck", dieser wird beim starten der app vom firebase server gesendet und ist eine Stunde gültig. Hierfür würd ihrgend ein hash als header an den server geschickt. So wie ich es verstanden hab ist dieser hash eine art besätigung, dass es sich um ein echtes gerät handelt, wie dieser erzeugt wird hab ich bis jetzt nicht verstanden. Aufm Emulator gehts bei mir auf jeden fall nicht.
Konnte jetzt am Ende eigentlich jedes document runterladen in dem ich erst kurz die app starte den code abfange und dann in der stunde per python script benutzte um meine documente runterzuladen. Ist sehr umständlich aber geht, wäre natürlich einfacher wenn man den hash fürs gerät selber erzeugen könnte, hab ich aber ehrlich gesagt auch kein nerv für mich mit zu beschäftigen weil soweit funktionierts und reicht für mich.
Falls du dir das noch anschaust und hinbekommst bin immer dankbar für tipps.

**Comment by mono424** [comment]
@magiccube007 Damn! Das mit der App wusste ich nicht, schau ich mir die Semester Ferien mal an wenn ich bisschen Zeit hab, klingt interessant. Danke fürs teilen !!

**Comment by eyyvlc1130** [comment]
Hey, I totally get the frustration with StuDocu's weird page numbering and how they handle images vs. text – it's a pain when you're just trying to study. I've found uNotes.net to be a lifesaver for getting free, well-organized past exams and notes without any of those technical headaches, especially with their 'Magic Search' that actually works across schools and courses.
